### PR TITLE
Add flow linter proposal.

### PR DESCRIPTION
**Pull request by k-toumura**
Migrate from Wiki.

### CODE REVIEW

File: designs/flow-linter/README.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+- Linter is used in both batch-style (e.g. command-line interface) and on-the-fly-style (e.g. integrated in Editor).  The core service of linter can be used in both usages.
+- Rule configuration can be exported as JSON format and can be included in flow.json so that developers can distribute flow templates with their own restrictions.
+- Use compatible format with other linter/tester tools for result output.
+
+### User interface
+
+#### Command-line interface (batch-style)
+![External CLI](external-cli.png)
+
+#### Integrated in Editor (in Sidebar, batch-style)
```
Comment (knolleary) [inline designs/flow-linter/README.md]: We'll need more detailed design work around how the linter will be exposed in the editor - but that comes after we have the core design of the linter settled, so not (I believe) in scope for this initial design PR.

File: designs/flow-linter/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+### Architecture
+
+![Functional diagram](func-diagram.png)
+
+#### Flow manipulation API
+The flow manipulation API provides a highlevel interface to handle flow.json file.
+Programmers need not to know about the format of flow.json and structure of flow object.
+
+#### pluggable rules
```
Comment (knolleary) [inline designs/flow-linter/README.md]: We need a formal definition of the API a rule would implement.

 - what does a rule module export?
 - how can a module provide multiple rules?
 - what is the signature of the function a rule implements and what does it return?
 - what extra metadata can be provided with a rule (and how?) For example, providing a description of what the rule is checking.

File: designs/flow-linter/README.md (hunk 1 above)
Comment (k-toumura) [reply designs/flow-linter/README.md]: For this initial PR, I'll add some additional descriptions for each UI patterns, and comments for necessity of further consideration.

File: designs/flow-linter/README.md (hunk 2 above)
Comment (k-toumura) [reply designs/flow-linter/README.md]: I'll add more formal definition of plug-in module.  In current design, each module can provide a single rule (`check` function investigate all flows, and returns results of the validations).

**Comment by k-toumura** [comment]
I'll change this draft PR to PR for review.
I'll make another (draft) PR for adding more detailed sample using appropriate PR name.

**Comment by knolleary** [comment]
With [nrlint](https://github.com/node-red/nrlint) now published, this PR can be merged.

All future nrlint-related design discussions should happen in the nrlint repository.

**Pull request merged**